python/datafusion/substrait.py

```diff
@@ -15,9 +15,156 @@
 # specific language governing permissions and limitations
 # under the License.
 
+from __future__ import annotations
 
-from ._internal import substrait
+from ._internal import substrait as substrait_internal
 
+from typing import TYPE_CHECKING
 
-def __getattr__(name):
-    return getattr(substrait, name)
+if TYPE_CHECKING:
+    from datafusion.context import SessionContext
+    from datafusion._internal import LogicalPlan
+
+
+class plan:
```

Comment:
According to PEP8:

> According to PEP8: Class names must follow CamelCase naming (Each new word is capitalized without underscore)

Reply:
Added, but also keeping a `plan` version marked as deprecated.